samples/templates/default-azuredeploy-docker.json

```diff
@@ -241,9 +238,25 @@
                     "type": "config",
                     "dependsOn": [
                         "[variables('appServiceResourceId')]",
-                        "[if(variables('deployAppInsights'),concat('Microsoft.Insights/components/', variables('appInsightsName')),resourceId('Microsoft.KeyVault/vaults', variables('serviceName')))]"
+                        "[if(variables('deployAppInsights'),concat('Microsoft.Insights/components/', variables('appInsightsName')),resourceId('Microsoft.KeyVault/vaults', variables('serviceName')))]",
+                        "[resourceId('Microsoft.KeyVault/vaults/secrets', variables('serviceName'), 'CosmosDb--Host')]"
```

Comment:
Addresses the issue where the container starts up before the keyvault secrets exist

Reply:
Does this change also need to be in the non-docker deployment script?

Reply:
Possibly, I haven't done enough testing with the other script yet